scripts/setup

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+# this shell script is responsible for:
+# - making sure your environment is setup, and
+# - setting up a database environment locally for you to use
+set -e
+
+usage() {
+  echo "usage: $0 [-h|--help]"
+  echo ""
+  echo "[-h|--help]      print this help text"
+}
+
+# script variables and defaults
+source $(dirname $0)/init.sh
+
+# the list of tools required to be installed to interact with this repo
+# this is meant to verify what is also in the ./README.md#environment-setup section
+ROOT_DIR="$(cd "$(dirname "$0")/.."; pwd)"
+
+TOOLS=(java docker)
+
+# process command-line options (if any)
+die() { log_error "$*" >&2; echo ""; usage; exit 2; }  # complain to STDERR and exit with error
+needs_arg() { if [ -z "$OPTARG" ]; then die "No arg for --$OPT option"; fi; }
+
+while getopts h-: OPT; do  # allow -h and -- "with arg"
+  # support long options: https://stackoverflow.com/a/28466267/519360
+  if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
+    OPT="${OPTARG%%=*}"       # extract long option name
+    OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
+    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+  fi
+  case "$OPT" in
+    h | help )     usage; exit 0 ;;
+    \? )           usage; exit 2 ;;  # bad short option (error reported via getopts)
+    * )            die "Illegal option --$OPT" ;;            # bad long option
+  esac
+done
+shift $((OPTIND-1)) # remove parsed options and args from $@ list
+
+
+# process positional arguments (if any)
+
+MSG_PREFIX="[$(basename $SCRIPTS_DIR)/$(basename $0)]"
+log() {
+  log_info "$MSG_PREFIX ${@}"
+}
+
+verify_tool() {
+  if ! command -v $1 &> /dev/null; then
+      log "'$1' could not be found"
+      exit 1
+  else
+    log "verified: ${@}"
+  fi
+}
```

Comment:
while reviewing this code ... I realized that @mflinn-broad added a `utils.sh`-method to do this (along with logging) - 🤓 

I updated the terra-java-project-template with the [changes in this commit](https://github.com/DataBiosphere/terra-java-project-template/pull/23/commits/046af5a034a33a59c8c75bcd0eba78fe952f9129)